apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -81,9 +82,10 @@ type K8ssandraClusterCondition struct {
 
 // K8ssandraStatus defines the observed of a k8ssandra instance
 type K8ssandraStatus struct {
-	Cassandra *cassdcapi.CassandraDatacenterStatus `json:"cassandra,omitempty"`
-	Stargate  *stargateapi.StargateStatus          `json:"stargate,omitempty"`
-	Reaper    *reaperapi.ReaperStatus              `json:"reaper,omitempty"`
+	Cassandra          *cassdcapi.CassandraDatacenterStatus `json:"cassandra,omitempty"`
+	CassandraTelemetry *telemetryapi.TelemetryStatus        `json:"cassandraTelemetry,omitempty"`
```

Comment:
Caution here: currently if you add a new field to the `K8ssandraStatus` struct, this has potential impacts in all controller methods called `setStatusForX` and `removeXStatus`. You will need to modify all of them. I know this isn't ideal, we might want to eventually merge all these methods into one single method that updates all fields at once.

Reply:
I think I'm just going to have to suck that up regrettably.